packages/grant-explorer/src/features/discovery/RoundMatchAmountBadge.tsx

```diff
@@ -0,0 +1,41 @@
+import { formatUnits, getAddress, zeroAddress } from "viem";
+import { Badge } from "../common/styles";
+import { useToken } from "wagmi";
+import { getPayoutToken } from "../api/utils";
+import { ChainId } from "common";
+
+type RoundCardStatProps = {
+  matchAmount: string;
+  token: string;
+  chainId: ChainId;
+};
+
+const formatter = new Intl.NumberFormat();
+
+export function RoundMatchAmountBadge({
+  chainId,
+  matchAmount,
+  token,
+}: RoundCardStatProps) {
+  const { data } = useToken({
+    address: getAddress(token),
+    chainId: Number(chainId),
+    enabled: Boolean(token && token !== zeroAddress),
+  });
+
+  const nativePayoutToken = getPayoutToken(token, chainId);
+
+  const symbol = data?.symbol ?? nativePayoutToken?.name ?? "ETH";
+  const decimals = data?.decimals ?? nativePayoutToken?.decimal ?? 18;
```

Comment:
Deciding what is the default payout token when a chain doesn't define one seems like something way above the responsibility of a badge component. :) `getPayoutToken` would probably be a better candidate for it.

Reply:
The data is fetched in useToken. We could move this into a usePayoutToken hook to handle this if that makes sense.
Otherwise we need to update getPayoutToken to receive a fallback token as an argument (data).